src/OpenTelemetry/Metrics/MeterProviderSdk.cs

```diff
@@ -74,9 +74,18 @@ internal sealed class MeterProviderSdk : MeterProvider
                     this.reader = new CompositeMetricReader(new[] { this.reader, reader });
                 }
 
-                if (reader is BaseExportingMetricReader baseExportingMetricReader)
+                if (reader is PeriodicExportingMetricReader periodicExportingMetricReader)
+                {
+                    exportersAdded.Append(((PeriodicExportingMetricReader)reader).Exporter);
```

Comment:
I think this can be simplified.
Line 77, the if statement uses the `is` operator to cast.
Lines 79 and 81 both perform the explicit conversion.

Seems like a preexisting issue, the same pattern is on lines 85 and 88.

Reply:
```suggestion
                    exportersAdded.Append(periodicExportingMetricReader.Exporter);
```